src/instance_info.rs

```diff
@@ -0,0 +1,171 @@
+use serde::{Deserialize, Serialize};
+use time::OffsetDateTime;
+
+// This is the local static that is intialized at runtime (technically at
+// the first request *to the instance-info endpoint) and contains the data
+// retrieved from the instance-info endpoint.
+pub(crate) static INSTANCE_INFO: Lazy<InstanceInfo> = Lazy::new(InstanceInfo::new);
+
+/// Handles instance info endpoint
+pub async fn instance_info(req: Request<Body>) -> Result<Response<Body>, String> {
+	// This will retrieve the extension given, or create a new string - which will
+	// simply become the last option, an HTML page.
+	let extension = req.param("extension").unwrap_or(String::new());
+	let response = match extension.as_str() {
+		"yaml" => info_yaml(),
+		"txt" => info_txt(),
+		"json" => info_json(),
+		"html" | _ => info_html(req),
+	};
+	response.map_err(|err| format!("{err}"))
+}
+
+fn info_json() -> Result<Response<Body>, Error> {
+	let body = serde_json::to_string(&*INSTANCE_INFO).unwrap_or("Error serializing JSON.".into());
+	Response::builder().status(200).header("content-type", "application/json").body(body.into())
+}
+
+fn info_yaml() -> Result<Response<Body>, Error> {
+	let body = serde_yaml::to_string(&*INSTANCE_INFO).unwrap_or("Error serializing YAML.".into());
+	// https://github.com/ietf-wg-httpapi/mediatypes/blob/main/draft-ietf-httpapi-yaml-mediatypes.md
+	Response::builder().status(200).header("content-type", "application/yaml").body(body.into())
+}
+
+fn info_txt() -> Result<Response<Body>, Error> {
+	Response::builder()
+		.status(200)
+		.header("content-type", "text/plain")
+		.body(INSTANCE_INFO.to_string(StringType::Raw).into())
+}
+fn info_html(req: Request<Body>) -> Result<Response<Body>, Error> {
+	let message = MessageTemplate {
+		title: String::from("Instance information"),
+		body: INSTANCE_INFO.to_string(StringType::Html),
+		prefs: Preferences::new(&req),
+		url: req.uri().to_string(),
+	}
+	.render()
+	.unwrap();
+	Response::builder().status(200).header("content-type", "text/html; charset=utf8").body(message.into())
+}
+#[derive(Serialize, Deserialize, Default)]
+pub(crate) struct InstanceInfo {
+	crate_version: String,
+	git_commit: String,
+	deploy_date: String,
+	compile_mode: String,
+	deploy_unix_ts: i64,
+	config: Config,
+}
+
+impl InstanceInfo {
+	pub fn new() -> Self {
+		Self {
+			crate_version: env!("CARGO_PKG_VERSION").to_string(),
+			git_commit: env!("GIT_HASH").to_string(),
+			deploy_date: OffsetDateTime::now_local().unwrap_or_else(|_| OffsetDateTime::now_utc()).to_string(),
+			#[cfg(debug_assertions)]
+			compile_mode: "Debug".into(),
+			#[cfg(not(debug_assertions))]
+			compile_mode: "Release".into(),
+			deploy_unix_ts: OffsetDateTime::now_local().unwrap_or_else(|_| OffsetDateTime::now_utc()).unix_timestamp(),
+			config: CONFIG.clone(),
+		}
+	}
+	fn to_table(&self) -> String {
+		let mut container = Container::default();
+		let convert = |o: &Option<String>| -> String { o.clone().unwrap_or("Unset".to_owned()) };
```

Comment:
```suggestion
		let convert = |o: &Option<String>| -> String { o.clone().unwrap_or("<i>(unset)</i>".to_owned()) };
```

To make it clearer the value is not set. I'd even recommend `<span>` and a CSS class to change the color of the text to something lighter to make it clear that the parameter is unset.